src/pins/state.rs

```diff
@@ -0,0 +1,67 @@
+//! Contains types that indicate pin states
+//!
+//! Please refer to [`Pin`] for documentation about how these types are used.
+
+use core::marker::PhantomData;
+
+/// Implemented by types that indicate pin state
+///
+/// [`Pin`] uses this type as a trait bound for the type parameter that
+/// indicates the pin's state. This is done for the purpose of
+/// documentation, to show which states a pin can be in. Other than that,
+/// this trait should not be relevant to users of this crate.
+///
+/// [`Pin`]: ../struct.Pin.html
```

Comment:
I think this link isn't valid anymore

Reply:
(also applies to all the other links to `Pin`)

Reply:
It is actually, due to the public re-export in `pins`. That's what I think anyway, but the link does work in the generated output.